### PR TITLE
Oleksandr Bunin COmmit

### PR DESCRIPTION
**Pull request by Sasha192**

### CODE REVIEW

File: sevntu_checks.xml [hunk 1]
```diff
@@ -0,0 +1,256 @@
+<?xml version="1.0"?>
```
Comment (Sofasmile) [inline sevntu_checks.xml]: add `target` folder in .gitignore

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -5,15 +5,33 @@
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
-    public int[] getDiagonal(int[][] matrix) {
-        return null;
+    protected int[] getDiagonal(int[][] matrix) {
+        int size = matrix[0].length > matrix.length ? matrix.length : matrix[0].length;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need this checking?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -5,15 +5,33 @@
...
+        int[] retArray = new int[size];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: retArray? what does it mean?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -5,15 +5,33 @@
...
+        for (int i = 0; i < size; i++) {
+            retArray[i] = matrix[i][i];
+        }
+        return retArray;
     }
 
-    public int getMin(int[][] matrix) {
-        return 0;
+    protected int getMin(int[][] matrix) {
+        int min = Integer.MAX_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why we use Integer.MAX_VALUE?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -5,15 +5,33 @@
...
+                min = min > matrix[i][j] ? matrix[i][j] : min;
+            }
+        }
+        return min;
     }
 
-    public int getMax(int[][] matrix) {
-        return 0;
+    protected int getMax(int[][] matrix) {
+        int max = Integer.MIN_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: the same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (Sasha192) [reply src/main/java/core/basesyntax/MatrixOperation.java]: > why we use Integer.MAX_VALUE?
 why not ? efficiency the same

Comment (Sasha192) [reply src/main/java/core/basesyntax/MatrixOperation.java]: > why we use Integer.MAX_VALUE?


why not ? efficiency the same

**Pull request closed without merging**